opba-protocols/hbci-protocol-tests/hbci-bdd-sandbox/src/test/java/de/adorsys/opba/protocol/hbci/tests/e2e/sandbox/HbciSandboxPaymentE2EHbciProtocolTest.java

```diff
@@ -0,0 +1,105 @@
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+
+/**
+ * Happy-path payment test that uses HBCI Sandbox to drive banking-protocol.
+ */
+@Transactional(propagation = Propagation.NOT_SUPPORTED)
+@SpringBootTest(classes = {
+        HbciProtocolApplication.class,
+        HbciServerApplication.class,  // Starting HBCI server within test so that application basically communicates with itself
+        HbciJGivenConfig.class
+}, webEnvironment = RANDOM_PORT)
+@ActiveProfiles(profiles = {ONE_TIME_POSTGRES_RAMFS, MOCKED_SANDBOX, HBCI_SANDBOX_CONFIG})
+class HbciSandboxPaymentE2EHbciProtocolTest extends SpringScenarioTest<
+        HbciServers,
+        HbciPaymentInitiationRequest<? extends HbciPaymentInitiationRequest<?>>,
+        HbciPaymentInitiationResult<? extends HbciPaymentInitiationResult<?>>
+    > {
+
+    private final String OPBA_LOGIN = UUID.randomUUID().toString();
+    private final String OPBA_PASSWORD = UUID.randomUUID().toString();
+
+    @LocalServerPort
+    private int port;
+
+    @Autowired
+    private HbciAdapterProperties adapterProperties;
+
+    @Autowired
+    private BankProfileJpaRepository bankProfileJpaRepository;
+
+    // TODO: Those dependencies do not need to be mocked, but should be optional
+    // Stubbing out xs2a protocol declared dependencies:
+    @MockBean
+    @SuppressWarnings("PMD.UnusedPrivateField") // Used to make Spring happy
+    private DtoMapper<Set<ValidationIssue>, Set<ValidationError>> dtoMapper;
+
+    // See https://github.com/spring-projects/spring-boot/issues/14879 for the 'why setting port'
+    @BeforeEach
+    @Transactional
+    void setBaseUrl() {
+        makeHbciAdapterToPointToHbciMockEndpoints();
+    }
+
+    @Test
+    void testSinglePaymentWithSca() {
+        given()
+                .rest_assured_points_to_opba_server()
+                .user_registered_in_opba_with_credentials(OPBA_LOGIN, OPBA_PASSWORD);
+
+        when()
+                .fintech_calls_single_payment_for_max_musterman_for_blz_30000003()
+                .and()
+                .user_logged_in_into_opba_as_opba_user_with_credentials_using_fintech_supplied_url(OPBA_LOGIN, OPBA_PASSWORD)
+                .and()
+                .user_max_musterman_provided_initial_parameters_to_make_payment()
+                .and()
+                .user_max_musterman_provided_password_to_embedded_authorization()
+                .and()
+                .user_max_musterman_selected_sca_challenge_type_push_tan_to_embedded_authorization()
+                .and()
+                .user_max_musterman_provided_sca_challenge_result_to_embedded_authorization_and_sees_redirect_to_fintech_ok_pis();
+        then()
+                .open_banking_has_consent_for_max_musterman_payment()
+                .fintech_calls_consent_activation_for_current_authorization_id();
+        // TODO check payment status
```

Comment:
Obviously needs to be finished :)

Reply:
should I change it to anonymous payment a we don't use consent save in hbci?
And also I don't know how to check status in hbci?

Reply:
As current HBCI sandbox does not store payment in the database, you need to check that OPBA table consent or (new payment table) has payment itself confirmed. We need GetPaymentStatus on HBCI and to store payment in sandbox database, but it is separate ticket